nwhacks/nwhacks2019
# Web/hacker application/input fields

:tickets: **Ticket(s)**: Closes #49 

---

## :construction_worker: Changes

* Adds some text input fields into hacker application pages.
* Also added a common prop type object that reinforces the structure of `hackerApplication`.
* Change `TextInput`'s `onChange` function to return the actual text instead of event target stuff.

## :thought_balloon: Notes

Page design doesn't have to follow design yet - just gonna focus on getting the logic ready and saving the application in Firestore first.

## :flashlight: Testing Instructions

* Make sure you're logged out
* Go to `application/hacker`
* Fill in the field in `First Name` in first page and `Second Name` in second page.
* refresh the page
- [ ] the values are saved
* Go to another page and come back to hacker application
- [ ] the values are saved
* Login as `newuser@example.com` with `testtest`
* Go to `application/hacker`
- [ ] the values are cleared
* Fill in the fields
* Logout
* Got back to hacker application
- [ ] the values are cleared
